bricks/template/__brick__/{{project_name.snakeCase()}}/lib/di/di.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:get_it/get_it.dart';
+import 'package:injectable/injectable.dart';
+
+import 'di.config.dart';
+
+final GetIt getIt = GetIt.instance;
+
+@injectableInit
+Future<void> configureInjection() async => getIt.init();
```

Comment:
@doannimble I think you forget to call this function in `main.dart` file:

```dart
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await FlutterConfig.loadEnvVariables();
  await configureInjection(); // here
  runApp(MyApp());
}
```

Reply:
Indeed. We don't yet have an implementation of `MainViewModel` on an actual screen, such as `HomeScreen`, so the issue did not occur.

Reply:
Should we add a basic implementation of MainViewModel on HomeScreen to verify the necessary initialization? Also, rename `MainViewModel` to `HomeViewModel`?

```
final homeViewModelProvider =
    StateNotifierProvider.autoDispose<MainViewModel, MainViewState>((ref) {
  return MainViewModel(
    getIt.get<GetUsersUseCase>(),
  );
});

class HomeScreen extends ConsumerStatefulWidget {
  const HomeScreen({super.key});

  @override
  HomeScreenState createState() => HomeScreenState();
}

class HomeScreenState extends ConsumerState<HomeScreen> {

  @override
  void initState() {
    super.initState();
    ref.read(homeViewModelProvider.notifier).getUsers();
  }
```

Reply:
Additionally, a ProviderScope is required

```
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await FlutterConfig.loadEnvVariables();
  await configureInjection();
  runApp(
    ProviderScope(
      child: MyApp(),
    ),
  );
}
```

<img src="https://github.com/nimblehq/flutter-templates/assets/16315358/b65c6ce8-533b-4ff2-bb26-1474aaeca261" width=200 />

Reply:
With those missing things ☝️  added, it works properly now.

env for testing
```
REST_API_ENDPOINT=https://jsonplaceholder.typicode.com/
```

![image](https://github.com/nimblehq/flutter-templates/assets/16315358/5740f1c8-e6b9-4848-8204-8af02a0d8a05)

Reply:
@luongvo I would suggest adding `riverpod` stuff in another ticket since the `StateNotifier` is very old now, and with the new riverpod's version, we can apply the new provider `AsyncNotifier` to replace `StateNotifier` completely IMO.
Ref:
- https://codewithandrea.com/articles/flutter-riverpod-async-notifier/
- https://github.com/nimblehq/internal-certification-flutter/blob/main/kaung/lib/screens/login/login_view_model.dart

Reply:
@manh-t Thanks for your information. We still need to complete the current state to ensure correct implementation. 

We can update to use the latest in a separate ticket. Maybe when working on a task from this RFC https://github.com/nimblehq/flutter-templates/discussions/202. Let's me note this down 👍

Reply:
> Should we add a basic implementation of `MainViewModel` on `HomeScreen` to verify the necessary initialization? Also, rename `MainViewModel` to `HomeViewModel`?

I think we don't need to add a basic implementation of `MainViewModel` on `HomeScreen` because that's just an example of this template. It's nice to have. We can do it in other PR. 

For this PR, as we want to add `full DI configuration`, so we must add `configureInjection()` in `main()` to fulfill the requirement.

> Additionally, a ProviderScope is required

I agree that we should add `ProviderScope` too.

Reply:
All fixed in https://github.com/nimblehq/flutter-templates/pull/244/commits/41c5114959cabf1994ed7b719c614d574561ffd5